feat(sourcemaps): Add more sourcemaps docs (#6667)
getsentry/sentry-docs: src/platform-includes/sourcemaps/artifact-and-release-bundles/javascript.mdx

> This article explains the differences between Artifact and Release Bundles, and the implications of using them. Visit [Uploading Source Maps](/platforms/javascript/guides/react/sourcemaps/uploading/) if you're looking for our guides on how to upload source maps.
> 

Reviewer: ```suggestion
This article explains the difference between Artifact and Release bundles, and the implications of using them. Visit [Uploading Source Maps](/platforms/javascript/guides/react/sourcemaps/uploading/) if you're looking for our guides on how to upload source maps.

```

In practice, matching the paths for artifacts discovery is prone to mistmatchs that might be caused by a wrong configuration of your bundler, CI/CD pipeline...

## Artifact Bundles

### What Are They?

An artifact bundle is our new bundle format that aims to address the shortcomings of release bundles by **adding the support for Debug IDs** and **removing the need to create a `release`**.

| Pros | Cons |
|:--|:--|
| With the use of Debug IDs, configuring your build pipeline to upload artifacts is trivial | Requires a version update to Javascript SDKs, bundler plugin ( and `sentry-cli`) |
| Associating a release is still possible, but not required | May cause problems with integrity checks

### How Do They Work?

```
Diagram Here
```

> Artifact Bundles solve the challenge of matching paths by using a new identification mechanism known as Debug IDs. Rather than relying on paths we uniquely identify and bind pairs of minified sources and source maps with an Debug ID(UUID). With this, we are able to quickly identify the minified source and corresponding source map without ever needing to check any paths. Thus, you don't have to worry about wrong paths configurations.
> 
> <Note>
> 
> If you want to learn more about the rationale behind Debug IDs, we suggest taking a look at our [engineering blog](https://sentry.engineering/blog/the-case-for-debug-ids).
> 
> </Note>

Reviewer: ```suggestion
Artifact bundles solve the challenge of matching paths by using a new identification mechanism known as Debug IDs. So rather than having to uniquely identify paths then bind pairs of minified sources and source maps with a Debug ID(UUID), you can quickly identify the minified source and corresponding source map without ever needing to check any paths. Thus, incorrect path configurations no longer present a problem.

<Note>

To learn more about the rationale behind Debug IDs, check out our [engineering blog](https://sentry.engineering/blog/the-case-for-debug-ids).

</Note>
```

> ### Associating a Release with Artifact Bundles

Reviewer: ```suggestion
### Associating a Release With Artifact Bundles
```

> Since you might still want to know to which release a specific artifact bundle is connected to, we designed a new way to still associate a `release` to your bundle.

Reviewer: ```suggestion
We designed a new way to associate a `release` with a bundle for those who still want to know which release a specific artifact bundle is connected to. This new kind of association to a `release` and optionally `dist` is known as a weak release association. It **doesn't require you to create of a `release`** before uploading sourcemaps, making it possible to create a `release` as a separate step down the pipeline.

```

> The new Artifact Bundle format supports a new kind of association to a `release` and optionally `dist`, known as weak release association. This type of association **will not require the creation of a `release`** before uploading sourcemaps and will consequentely allow the creation of a `release` as a separate step down the pipeline.
> 
> With an associated `release` and optionally `dist` you will be able to quickly go to the Artifact Bundle from your `release` in Sentry, without having to worry about which artifact bundle was used for your errors.
> 

Reviewer: ```suggestion

With an associated `release` and optionally `dist` you'll be able to quickly go to the artifact bundle from your `release` in Sentry, without having to worry about which artifact bundle was used for your errors.

```

> ## Summary
> - We recommend you try Artifact Bundles with Debug IDs, if you've failed to upload source maps in the past.
> - However, we recommend you continue to use Release Bundles if your application already uses them successfully.

Reviewer: ```suggestion
## Summary
If you've had trouble uploading source maps in the past, we recommend you try artifact bundles with Debug IDs, but if your application already uses them successfully, we recommend you continue to use release bundles .
```
